codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/PaginatorGenerator.kt

```diff
@@ -141,8 +141,9 @@ class PaginatorGenerator private constructor(
 
                 /// Create the pagination stream
                 ///
-                /// _Note:_ No requests will be dispatched until the stream is used (eg. with [`.next().await`](tokio_stream::StreamExt::next)).
-                pub fn send(self) -> impl #{Stream}<Item = #{item_type}> + #{Unpin} {
+                /// _Note:_ No requests will be dispatched until the stream is used
+                /// (e.g. with [`.next().await`](aws_smithy_async::future::fn_stream::FnStream::next)).
+                pub fn send(self) -> #{fn_stream}::FnStream<#{item_type}> {
```

Comment:
hmmm...not sure I like this. It's pretty constraining and also exposes a weird name to customers. I think we should wrap this in a newtype so we aren't publicly exposing the `FnStream` internals

Reply:
> Can we make this PR much smaller to start piecing off individual chunks?

We can create a smaller PR just to remove `impl Stream` from `aws-smithy-async` and but keep it for event streaming (that'll take care of Burak & David's questions around `FnStream` since the code around event stream will be unchanged).

The above `send` methods needs to return a new-type hiding `FnStream`.